### PR TITLE
feat(ui): Documentation Editor Improvements

### PR DESCRIPTION
**Pull request by jjoyce0510**
## Summary

A few features we added in this PR:

1. Full-Screen Documentation Preview
2. Persisted "last location" in the browser, so that opening a new tab seamlessly drops user back into context

The combination of these 2 provides a pretty awesome documentation editing experience.

## Status
Ready for review

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx [hunk 1]
```diff
@@ -135,22 +133,19 @@ export const DescriptionEditor = ({ onComplete }: { onComplete?: () => void }) =
...
-                </Button>
-                <Button onClick={handleSaveDescription} disabled={!isDescriptionUpdated}>
-                    <CheckOutlined /> Save
-                </Button>
-            </TabToolbar>
+            <DescriptionEditorToolbar
+                onSave={handleSave}
+                onClose={handleConfirmClose}
+                disableSave={!isDescriptionUpdated}
+            />
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/entity/shared/tabs/Documentation/components/DescriptionEditor.tsx:140]: nice!

**Review by chriscollins3456 — APPROVED** [review]
looking good to me! i have a couple of nits so feel free to take or leave them

File: datahub-web-react/src/app/entity/shared/tabs/Documentation/DocumentationTab.tsx [hunk 2]
```diff
@@ -36,15 +37,19 @@ export const DocumentationTab = ({ properties }: { properties?: Props }) => {
...
     const isEditing = queryString.parse(useLocation().search, { parseBooleans: true }).editing;
+    const showModal = queryString.parse(useLocation().search, { parseBooleans: true }).modal;
 
     useEffect(() => {
         const editedDescriptions = (localStorageDictionary && JSON.parse(localStorageDictionary)) || {};
         if (editedDescriptions.hasOwnProperty(urn)) {
-            routeToTab({ tabName: 'Documentation', tabParams: { editing: true } });
+            routeToTab({
+                tabName: 'Documentation',
+                tabParams: { editing: true, modal: (showModal && true) || false },
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Documentation/DocumentationTab.tsx:47]: oooo!

File: datahub-web-react/src/app/context/UserContextProvider.tsx [hunk 3]
```diff
@@ -124,6 +129,41 @@ const UserContextProvider = ({ children }: { children: React.ReactNode }) => {
         }
     }, [state, localState.selectedViewUrn, setDefaultSelectedView]);
 
+    /**
+     * Route to the most recently visited path once on first load of home page, if present in local storage.
+     */
+    useEffect(() => {
+        if (
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/context/UserContextProvider.tsx]: done!

**Pull request merged**